qualtran/_infra/adjoint.py

```diff
@@ -158,9 +159,16 @@ def adjoint(self) -> 'Bloq':
         """The 'double adjoint' brings you back to the original bloq."""
         return self.subbloq
 
-    def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
+    def build_call_graph(self, ssa: 'SympySymbolAllocator') -> 'BloqCountDictT':
         """The call graph takes the adjoint of each of the bloqs in `subbloq`'s call graph."""
-        return {(bloq.adjoint(), n) for bloq, n in self.subbloq.build_call_graph(ssa=ssa)}
+        sub_cg = self.subbloq.build_call_graph(ssa=ssa)
+        if isinstance(sub_cg, set):
+            counts = Counter['Bloq']()
+            for bloq, n in sub_cg:
+                counts[bloq.adjoint()] += n
+            return counts
+        else:
+            return {bloq.adjoint(): n for bloq, n in sub_cg.items()}
```

Comment:
Should we be worried about a situation where different bloqs return the same `adjoint()`?  I think it should never happen, but am not so sure. (same concern for `bloq.controlled()`).

@mpharrigan @tanujkhattar?

Reply:
that's an interesting corner case to consider; and the type of thing we need to be concerned about now that we are being strict about having only unique bloqs in the callees. I don't _think_ it should happen. I defer to @dstrain115 on whether he thinks it's worth defending against that vs blocking progress on the overall migration

Reply:
I've added the Counter for both sets and dicts, which should alleviate the concern.  Will push tomorrow unless there are any further issues.